File: Leetcode/1781.go
func beautySum(s string) int {
	const MAXN int = 26
	const LENGTH int = 502
	var pre [LENGTH][MAXN]int
	for i := 1; i <= len(s); i++ {
		for j := 0; j < i; j++ {
			pre[i][s[j] - 'a']++
		}
	}
	var diff [MAXN]int
	var maxFreq, minFreq int
	var sum = 0
	for r := 1; r <= len(s); r++ {
		for l := 1; l <= r; l++ {
			for k := 0; k < MAXN; k++ {
				diff[k] = pre[r][k]-pre[l-1][k]
			}
			maxFreq = 0
			for k := 0; k < MAXN; k++ {
				if maxFreq < diff[k] {
					maxFreq = diff[k]
				}
			}
			if maxFreq > 0 {
				minFreq = LENGTH
				for k := 0; k < MAXN; k++ {
					if diff[k] != 0 && minFreq > diff[k] {
						minFreq = diff[k]
					}
				}
				sum += maxFreq - minFreq
			}
		}
	}
	return sum
}

File: Leetcode/1781/1781.cpp
constexpr int MAXN = 26;
constexpr int LENGTH = 505;

class Solution {
public:
    int pre[LENGTH][MAXN];
    int beautySum(string s) {
        memset(pre, 0, sizeof pre);
        int n = (int)s.size();
        for (int i = 1; i <= n; ++i) {
            for (int j = 0; j < i; ++j) {
                ++pre[i][s[j] - 'a'];
            }
        }
        int diff[MAXN];
        int maxFreq, minFreq;
        int sum = 0;
        for (int r = 1; r <= n; ++r) {
            for (int l = 1; l <= r; ++l) {
                for (int k = 0; k < MAXN; ++k) {
                    diff[k] = pre[r][k] - pre[l-1][k];
                }
                maxFreq = 0;
                for (int k = 0; k < MAXN; ++k) {
                    maxFreq = max(maxFreq, diff[k]);
                }
                if (maxFreq > 0) {
                    minFreq = LENGTH;
                    for (int k = 0; k < MAXN; ++k) {
                        if (diff[k] != 0) {
                            minFreq = min(minFreq, diff[k]);
                        }
                    }
                    sum += maxFreq - minFreq;
                }
            }
        }
        return sum;
    }
};